crates/bitwarden/src/crypto/aes_ops.rs

```diff
@@ -39,16 +39,20 @@ pub fn decrypt_aes256_hmac(
     decrypt_aes256(iv, data, key)
 }
 
-pub fn encrypt_aes256(
+pub fn encrypt_aes256(data_dec: &[u8], key: GenericArray<u8, U32>) -> Result<EncString> {
+    let mut iv = [0u8; 16];
+    rand::thread_rng().fill_bytes(&mut iv);
+    let data = cbc::Encryptor::<aes::Aes256>::new(&key, &iv.into())
+        .encrypt_padded_vec_mut::<Pkcs7>(data_dec);
+
+    Ok(EncString::AesCbc256_B64 { iv, data })
+}
+
+pub fn encrypt_aes256_hmac(
```

Comment:
I guess we could now change this fn to something like this to avoid duplication: 
```rust
pub fn encrypt_aes256_hmac(
    data_dec: &[u8],
    mac_key: GenericArray<u8, U32>,
    key: GenericArray<u8, U32>,
) -> Result<EncString> {
    let enc = encrypt_aes256(data_dec, key)?;
    let EncString::AesCbc256_B64 { iv, data } = enc else {
        unreachable!()
    };
    let mac = validate_mac(&mac_key, &iv, &data)?;

    Ok(EncString::AesCbc256_HmacSha256_B64 { iv, mac, data })
}
```

Don't like the partial match very much though, what do you think?

Reply:
Feels like it's more pain than gain honestly. We could extract the previous operations to a separate function.

```rust
fn partial() {
    let mut iv = [0u8; 16];
    rand::thread_rng().fill_bytes(&mut iv);
    let data = cbc::Encryptor::<aes::Aes256>::new(&key, &iv.into())
        .encrypt_padded_vec_mut::<Pkcs7>(data_dec);
  (iv, data)
}